crates/ide/src/hover.rs

```diff
@@ -111,16 +113,69 @@ pub(crate) fn hover(
         kind if kind.is_trivia() => 0,
         _ => 1,
     })?;
-    let token = sema.descend_into_macros(token);
-    let node = token.parent()?;
+
+    let mut seen = HashSet::default();
+
+    let mut fallback = None;
+    sema.descend_into_macros_many(token.clone())
+        .iter()
+        .filter_map(|token| match token.parent() {
+            Some(node) => {
+                match find_hover_result(&sema, file_id, offset, config, token, &node, &mut seen) {
+                    Some(res) => match res {
+                        ControlFlow::Break(inner) => Some(inner),
+                        ControlFlow::Continue(_) => {
+                            if fallback.is_none() {
+                                // FIXME we're only taking the first fallback into account that's not `None`
+                                fallback = type_hover(&sema, config, &token)
+                                    .or(hover_for_keyword(&sema, config, &token));
```

Comment:
Whoops, I proposed the wrong order here, that should be
```suggestion
                                fallback = hover_for_keyword(&sema, config, &token)
                                    .or(type_hover(&sema, config, &token));
```
Otherwise when hovering a keyword we'll show the type of the expression it belongs to instead